sdk/storage/Azure.Storage.Blobs/tests/AppendBlobClientTests.cs

```diff
@@ -99,59 +99,59 @@ public void Ctor_CPK_EncryptionScope()
         [Test]
         public void WithSnapshot()
         {
-            var containerName = GetNewContainerName();
-            var blobName = GetNewBlobName();
-
-            BlobServiceClient service = GetServiceClient_SharedKey();
-
-            BlobContainerClient container = InstrumentClient(service.GetBlobContainerClient(containerName));
-
-            AppendBlobClient blob = InstrumentClient(container.GetAppendBlobClient(blobName));
-
-            var builder = new BlobUriBuilder(blob.Uri);
-
-            Assert.AreEqual("", builder.Snapshot);
-
-            blob = InstrumentClient(blob.WithSnapshot("foo"));
-
-            builder = new BlobUriBuilder(blob.Uri);
-
-            Assert.AreEqual("foo", builder.Snapshot);
-
-            blob = InstrumentClient(blob.WithSnapshot(null));
+            // Arrange
+            string accountName = "accountname";
+            string containerName = GetNewContainerName();
+            string blobName = "my/blob/name";
```

Comment:
Please also add other special chars besides "/" in the blob name like 
%:?*:"<>|

Reply:
That case is covered by tests that were previously written.  https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Blobs/tests/ContainerClientTests.cs#L2256

Reply:
Would you like add WithVersion, withSanpshot test in this case?